### ISSUE TITLE
old Libary out of packages directory used 

### ISSUE THREAD

**Issue by Alf-R** [Sloeber/arduino-eclipse-plugin#1536]
**Describe your environment**   
I use the Sloeber product Version 4.4.1    /  windows 10
   
The url of the json that you used to download the platform  
https://downloads.arduino.cc/packages/package_index.json
https://raw.githubusercontent.com/jantje/hardware/master/package_jantje_index.json
https://raw.githubusercontent.com/jantje/ArduinoLibraries/master/library_jantje_index.json
https://arduino.esp8266.com/stable/package_esp8266com_index.json
https://downloads.arduino.cc/libraries/library_index.json

A image of the project properties arduino showing the board settings   
![image](https://user-images.githubusercontent.com/20225457/209471139-44a204c6-9407-4dd4-8160-c13e89f106ad.png)

**Describe the bug**   
With the Board json (https://raw.githubusercontent.com/jantje/hardware/master/package_jantje_index.json) the board ist loaded and additional several libary's .
For example ArduinoOTA Version 1.0 -> shown in the libary's directory of the project.
With preferences -> libary manager I download and install ArduinoOTA Version 1.0.9 but this is not used for compile of the project.
Error report:
C:\prg\Sloeber\arduinoPlugin\packages\esp8266\hardware\esp8266\3.0.2\libraries\ArduinoOTA\ArduinoOTA.cpp:268:12: error: 'stopAll' is not a member of 'WiFiUDP'
  268 |   WiFiUDP::stopAll();
      |            ^~~~~~~
--> this is in the current version (1.0.9) not implemented

provide the name of the board and the platform version you are using.  
Board: ESP8266 (with ESP32 the same old libary is used but it works - same code- and libary implentation)

**Expected behavior**   
All libary's used as loaded in the libary manager

**Additional context**   
So basically there are severeal old libary's used for compile that brings up error's and issues to fight with.

cross check:
If I compile this code with Arduino IDE 1.8.13 it works fine ....

May be I just have to change some settings but I have no idea what and where

**Comment by jantje** [comment]
### I'm a bit confused here.
The board settings states you have a board installed from https://arduino.esp8266.com/stable/package_esp8266com_index.json
However you state 

> With the Board json (https://raw.githubusercontent.com/jantje/hardware/master/package_jantje_index.json) the board ist loaded and additional several libary's .

Both boards are unrelated.
This makes it very hard to provide specific info.

### About Expected behaviour
>All library's used as loaded in the library manager

This is not the implemented behaviour in Sloeber at all. (And not the behaviour described by the Arduino framework)
At compile time Sloeber uses the libraries that are in the libraries folder. You can add and remove libraries there via the menu "add libraries to project" or let Sloeber manage the content of the folder (default behaviour) or a mix of the 2 (adding manually the ones Sloeber does not add automatically). 
You can allow Sloeber to import libraries in your project in the workspace preferences (default on)
![afbeelding](https://user-images.githubusercontent.com/1175841/209474379-0613c6b0-758f-4328-8d78-847aef30031a.png)

**What triggers the library search**
Sloeber subscribes on indexer updates with CDT.
When CDT informs Sloeber of indexer updates Sloeber queries the CDT indexer for "unresolved includes". 

CDT reports unresolved includes as header files, assume one of the unresolved includes is "xx.h". Based on the CDT provided "xx.h"  Sloeber will search for the library with the name"xx". Sloeber assumes the library name "xx" to be the folder name containing the library code. Therefore Sloeber will search for a folder named "xx" in any of the provided places to look for libraries.
notes: 
1. The Arduino framework does not enforce the library xx to contain a xx.h file.  
2. Sloeber contains additional code to mitigate the Arduino framework discrepancy a bit (but far from fully)

**Here are the rules when automatically importing libraries to project in the libraries folder.**

1. If a Library is found in the libraries folder of the board core; this one is used (this is what I see in your case)
2. Else: If a Library part of the private libraries is found this one is used (if multiple private libraries contain the library the order of the listing of private libraries folders will decide on the library used)
3. Else: if a library installed by the library manager is found this one is used 
4. Else: if you have the option to "install missing libraries" on; the library manager will search for a matching library in the library jsons provided in the "third party index url's" and if a matching library is found the library manager will install it but not import it into the project (this is supposed-see later on the supposed- to happen at the next search for unresolved includes). 

Note that "the find the library" is different implementation then Arduino IDE. 
Therefore a successful "install missing library" may not lead to a find the library to "add to project" nor to a resolvement of the unresolved include.


**Manually importing libraries to a project**
Sloeber first searches all possible libraries to be added. Libraries are named based on the folder name containing the Library code.
In case of naming conflicts (that is 2 libraries having the same (folder)name the library that will be added to the project will be first found in the list below

1. The library part of the product
2. The Library part of the private libraries(the search is in order of the listing of private libraries)
3. The library installed by the library manager
